src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,166 @@
+    }
+
+    private class Node<K,V> implements Entry<K,V> {
+        private int hash;
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public final int hashCode() {
+            hash = 31;
+            hash = hash * 17 + key.hashCode();
+            return hash;
+        }
+
+        @Override
+        public String toString() {
+            return "Node{" + "hash=" + hash + ", key=" + key
+                    + ", value=" + value + ", next=" + next + '}';
+        }
+    }
+
+    private Node<K,V>[] resize() {
+        Node<K, V>[] oldTab = table;
+        int oldCap = (oldTab == null) ? 0 : oldTab.length;
+        int oldThr = threshold;
+        int newCap;
+        int newThr;
+        if (oldCap > 0) {
+            newCap = oldCap << 1;
+            newThr = oldThr << 1;
+        } else {
+            newCap = INIT_SIZE;
+            newThr = (int) (INIT_LOAD_FACTOR * INIT_SIZE);
+        }
+        threshold = newThr;
+        Node<K,V>[] newTab = (Node<K,V>[]) new Node[newCap];
+        if (oldTab != null) {
+            transfer(newTab);
+        }
+        table = newTab;
+        return newTab;
+    }
+
+    private void transfer(Node<K,V>[] newTab) {
+        Node<K,V>[] oldTab = table;
+        int oldCap = oldTab.length;
+        int newCap = newTab.length;
+        for (int j = 0; j < oldCap; ++j) {
+            Node<K,V> temp = oldTab[j];
+            if (temp != null) {
+                oldTab[j] = null;
+                do {
+                    Node<K,V> next = temp.next;
+                    int i = indexFor(temp.hash, newCap);
+                    temp.next = newTab[i];
+                    newTab[i] = temp;
+                    temp = next;
+                } while (temp != null);
+            }
+        }
+    }
+
+    Node<K,V> newNode(int hash, K key, V value, Node<K,V> next) {
+        return new Node<>(hash, key, value, next);
+    }
+
+    private Node<K,V> getNode(int hash, Object key) {
+        Node<K, V>[] tab = table;
+        Node<K, V> tabNode;
+        Node<K, V> temp;
+        int n;
+        K k;
+        if ((tab != null) && ((n = tab.length) > 0)
+                && (tabNode = tab[(n - 1) & hash]) != null) {
+            if (tabNode.hash == hash && ((k = tabNode.key) == key
+                    || (key != null && key.equals(k)))) {
+                return tabNode;
+            }
+            if ((temp = tabNode.next) != null) {
+                do {
+                    if (temp.hash == hash && ((k = temp.key) == key
+                            || (key != null && key.equals(k)))) {
+                        return temp;
+                    }
+                } while ((temp = temp.next) != null);
+            }
+        }
+        return null;
+    }
+
```

Comment:
<img width="681" alt="image" src="https://user-images.githubusercontent.com/81781479/207386433-0defa720-bc41-4d48-ac2f-ff8849029eb5.png">
why is your implementation identical to the original HashMap implementation?
try to implement it yourself - to understand how it works ))